src/commands/backup.rs

```diff
@@ -264,7 +280,9 @@ impl BackupCmd {
             }
 
             info!("backup of {source} done.");
+            run_after.run(&format!("run-after backup {source}"))?;
         }
+        config.backup.run_after.run("run-after backup")?;
```

Comment:
do we always want to run `run-after` even if somewhere fails after a `run-before`? 
in an analog way of `open` / `close` file handles?

Reply:
I think there are cases. For instance, if users want to mount a filesystem before backing it up - then they probably want to unmount mo-matter-what.
But actually this is also a weakness - if we catch an error in between, the `run-after` command won't even be executed...

I think adding additional hooky like `run-after-failed` or `run-after-successful` should be the solution here. I'll see if we can add them here..

Reply:
Should `run-after`, `run-after-success`, `run-after-failed` exist at the same time?
Or do you think `run-after-success`, `run-after-failed` can replace and exist without ~run-after~?

Reply:
I actually called them `run-failed` and `run-finally`.
- `run-after` is only called if everything was ok
- `run-failed` is only called if there was a failure
- `run-finally` is always called